x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalVerifier.java

```diff
@@ -20,20 +20,23 @@
 
 public final class LogicalVerifier extends PostOptimizationPhasePlanVerifier<LogicalPlan> {
 
-    public static final LogicalVerifier INSTANCE = new LogicalVerifier();
+    private LogicalVerifier(boolean isLocal) {
+        super(isLocal);
+    }
 
-    private LogicalVerifier() {}
+    public static LogicalVerifier getLocalVerifier() {
+        return new LogicalVerifier(true);
+    }
+
+    public static LogicalVerifier getGeneralVerifier() {
```

Comment:
I think so far we only have a "local" thing (verifier, mapper, optimizer etc.) and the unqualified other variant, which is the coordinator one. Let's avoid introducing a notion of a "general" thing (which wouldn't be "general", as it wouldn't apply locally :) ). 
We should use "coordinator" wherever we need to denote a non-local execution.